pkg/controller/runtime/adapter.go

```diff
@@ -338,11 +338,18 @@ func (adapter *adapter) Teardown(ctx context.Context, resourcePointer resource.P
 		return false, fmt.Errorf("resource %q/%q is not an output for controller %q, teardown attempted on %q", resourcePointer.Namespace(), resourcePointer.Type(), adapter.name, resourcePointer.ID())
 	}
 
-	return adapter.runtime.state.Teardown(ctx, resourcePointer, state.WithTeardownOwner(adapter.name))
+	var opts []state.TeardownOption
+
+	opOpt := controller.ToOptions(opOpts...)
+	if !opOpt.IgnoreOwner {
```

Comment:
this doesn't look like `IgnoreOwner`, but rather `WithOwner("")`, so I probably would rather use explicit `WithOwner`, even with empty owner